### PR TITLE
Real time data pt1

### PR DESCRIPTION
**Pull request by dasgarner**
Foundational work for data connectors and real time data.

 - DataSet extended with "Is Real Time" option and JavaScript
 - Scheduled extended with "Data Connector" event type and dataSetId/dataSetParams
 - XMDS extended to make both of the above available

### CODE REVIEW

File: views/schedule-form-add.twig [hunk 1]
```diff
@@ -136,7 +136,7 @@
 
                         {% set title %}{% trans "Layout" %}{% endset %}
                         {% set helpText %}{% trans "Please select a Layout for this Event to show" %}{% endset %}
-                        {{ forms.dropdown("campaignId", "single", title, campaign.campaignId, [campaign], "campaignId", "campaign", helpText, "layout-control", "", "", "", attributes, "", "", "", readonlySelect) }}
+                        {{ forms.dropdown("campaignId", "single", title, null, null, "campaignId", "campaign", helpText, "layout-control", "", "", "", attributes, "", "", "", readonlySelect) }}
```
Comment (PeterMis) [inline views/schedule-form-add.twig:139]: Could we double check if that breaks the add schedule form from Layout/Campaign pages please?
I think it relies on that being passed from the backend to populate the field (the layout/campaign field is also disabled in this flow)

Comment (dasgarner) [reply views/schedule-form-add.twig]: Yes you're right, reverting.

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Review by PeterMis — APPROVED** [review]

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Pull request merged**